### ISSUE TITLE
Mangafox manga no longer downloading

### ISSUE THREAD

**Issue by ABtheCoder** [NguyenDanPhuong/MangaRipper#95]
[2017-09-29.log](https://github.com/NguyenDanPhuong/MangaRipper/files/1344589/2017-09-29.log)
Manga from mangafox just stopped downloading giving me an error about an invalid URL. This happened quite recently as it worked before the 25th for sure. This is rather strange since I don't see any fundamental changes in mangafox or how it works so hopefully, this'll be a quick fix. Thanks

**Comment by JedBurke** [comment]
They're using the scheme which was used to download the page. Instead of specifying `http://path/to/manga` or `https://path/to/manga` for the chapter, they're using `//path/to/manga`. In other words, if the page was delivered via HTTP the new uri will be requested with HTTP. Browsers can infer the scheme, that's why it isn't an issue there. The URI constructor (used in MangaRipper) (probably) cannot, that's why there is an error here.

Adding `http:` where there is none does indeed appear to fix the issue.

**Comment by iYatsumi** [comment]
![image](https://user-images.githubusercontent.com/8799430/31039550-2a4145f4-a544-11e7-8868-9c23d9984756.png)

i dont know what you mean by that but all my links are with http:

**Comment by JedBurke** [comment]
The MangaFox page which lists the series' chapters no longer uses a scheme. You will see it when viewing the page's source or inspecting an element.

![capture](https://user-images.githubusercontent.com/6389344/31039699-17b8fefe-a580-11e7-92fd-ee29b419bda8.PNG)

Notice the value belonging to `href` in the highlighted section starting with `//mangafox.me` and not `http://mangafox.me`. Copying the chapter's link or URL from the URL bar to the clipboard (tested in Firefox) will insert the missing `http:`.

**Comment by JedBurke** [comment]
@iYatsumi 

Yes, the error seen with KissManga is indeed another issue. Please see closed issues 91 and 92 for more information. The currently open pull request is expected to address it.

I've also checked your log where similarly to ABtheCoder's log, the following line appears before the exception:
```
2017-09-29 19:30:23.1807 | ERROR | WorkerController | Failed to download chapter: //mangafox.me/manga/days/v15/c125/1.html 
```

**Comment by iYatsumi** [comment]
you mean like this or ? 

![image](https://user-images.githubusercontent.com/8799430/31046000-49446356-a5b6-11e7-8d0d-43f714ceb168.png)

because i dont quite understand what you are trying to refer at, i get that you inspected the page and that manga doesnt use the http thing anymore but what exactly it is that you want me to do about it ? If you could guide me it would be appreciated.

**Comment by JedBurke** [comment]
No. MangaRipper downloads the HTML of the chapter list where the links to individual chapters have the scheme (http:) removed and later tries to parse them. It's done automatically, you'll need to fix it in the application.

My thought for a solution is that the application is to insert the scheme after the chapter list has been downloaded but before the chapter itself begins downloading. Please see line 50 on the right in the proposed solution https://github.com/NguyenDanPhuong/MangaRipper/pull/96/commits/7935ff201db7e2b93f44dfbac2500bcbac6061d9

**Comment by Dichter68** [comment]
MangaRipper stopped downloading from Mangahere too. Every time I try to get the chapters I get an error:
![error](https://user-images.githubusercontent.com/25414685/31048777-2508cefe-a625-11e7-8cb7-8cd2eb01fda1.jpg)
Is it the same or a similar problem MangaRipper has with mangafox?

**Comment by JedBurke** [comment]
@Dichter68 MangaHere is doing the same as MangaFox when listing chapters on its website.

**Comment by JedBurke** [comment]
For those wanting to download from MangaFox and MangaHere, the following is a temporary fix:

Please [download](https://github.com/NguyenDanPhuong/MangaRipper/files/1347403/Plugins.zip) the zip file, [unblock](https://github.com/NguyenDanPhuong/MangaRipper/wiki/F.A.Q.#cannot-start-mangaripper-the-log-file-shows-nothing) ([alt](https://github.com/NguyenDanPhuong/MangaRipper/issues/50#issuecomment-282549931)) if necessary, and extract the contents into to your MangaRipper directory. Depending on your settings, you'll receive a prompt to overwrite files upon extraction.

**Comment by iYatsumi** [comment]
![image](https://user-images.githubusercontent.com/8799430/31058517-8e5003ce-a6ba-11e7-955e-c6680688c6ff.png)

this is what i get after extracting your .dll into the plugins folder. 

[2017-10-01.log](https://github.com/NguyenDanPhuong/MangaRipper/files/1347465/2017-10-01.log)

**Comment by JedBurke** [comment]
@iYatsumi Sorry about that. See my previous post for the updated link.

**Comment by iYatsumi** [comment]
it still gives me the same error

**Comment by JedBurke** [comment]
@iYatsumi The link's been updated again.

**Comment by iYatsumi** [comment]
tyvm it worked

**Comment by JedBurke** [comment]
No problem. I once again apologize for the back and forth.

**Comment by moax101** [comment]
How to solve this problem? is easiest way?
![image](https://user-images.githubusercontent.com/32432153/31273296-d7dadd1c-aa42-11e7-9999-f7ff4c3fff83.png)

**Comment by JedBurke** [comment]
@moax101 In the meantime, follow the instructions in my comment a little bit up. There is a zip file you must download and extract.

https://github.com/NguyenDanPhuong/MangaRipper/issues/95#issuecomment-333396599

**Comment by moax101** [comment]
@JedBurke im very sorry dude but i cant keep up
what zip file to download?

**Comment by moax101** [comment]
im not so good in this kind of problem sorry

**Comment by JedBurke** [comment]
@moax101 No worries.

[Download](https://github.com/NguyenDanPhuong/MangaRipper/files/1347642/Plugins.zip) this one. Right-click the file and "Properties". Click unblock. 

![unblock](https://user-images.githubusercontent.com/6389344/31284941-fbc5c45a-aaf4-11e7-9a30-b27b78834c1a.JPG)

Open the zip file, double-click "Plugins", and copy the files into to your MangaRipper Plugins directory.

![extract](https://user-images.githubusercontent.com/6389344/31285352-05422f68-aaf6-11e7-998e-dde21ba7fc37.JPG)

When you are done, restart MangaRipper.

**Comment by moax101** [comment]
ok thanks @JedBurke

**Comment by eksynn** [comment]
the temporary patch worked for me too ^^
case closed! (?)

**Comment by seiaxen** [comment]
Heya guys, to let you know, I follow the lastest comment by Jed in here, the error code is different. Could you guys help me? Thanks
![hmmm](https://user-images.githubusercontent.com/28289533/31585821-725964ec-b1f2-11e7-8faf-3c6eeb55229b.png)

**Comment by JedBurke** [comment]
@seiaxen Are you using version 1.0.264? Also please upload your log.

**Comment by seiaxen** [comment]
@JedBurke The one showing that error is the prev version (1.0.225) . When I paste the plugin files and unblocking via powershell, it still showing the parse error at MangaRipper.Core . And now I'm trying using my working pc (the previos post is on my personal one), after following this [web](https://blogs.technet.microsoft.com/heyscriptingguy/2012/09/14/easily-unblock-all-files-in-a-directory-using-powershell/) I can get the manga list. Can someone explain this? Thanks. (I'll try to update my personal one tonight, wml)

**Comment by JedBurke** [comment]
@seiaxen

Sorry for the late response, I'm glad you got it working. 

I'll try answering some of your previous questions.

> And now I'm trying using my working pc (the previos post is on my personal one), after following this web I can get the manga list. Can someone explain this?

The reason it works after unblocking it is because Windows tries to protect you from viruses by limiting what certain applications (which have been downloaded from the Internet) can do. If the plugin is blocked, it can't do its job nor can MangaRipper if it is also blocked.

If you're not comfortable with PowerShell, you can unblock your files directly from Windows Explorer. Just right-click the file and select `Properties`. In the window will be a button with the text `Unblock`. Since it will only appear when the file is blocked, it is also useful to check if your files *are* blocked.

![Blocked](https://user-images.githubusercontent.com/6389344/31284941-fbc5c45a-aaf4-11e7-9a30-b27b78834c1a.JPG)

Also, if you unblock a zip file all of the extracted files would be unblocked too. If not, the extracted files will remain blocked. See the FAQ for more.

As for your parse error. I've checked your log and it appears the server wasn't giving you the chapters due to the URI not having a trailing slash (/). Compare `http://www.mangahere.co/manga/<series-name>` to `http://www.mangahere.co/manga/<series-name>/`

This is what you (and I) got without the slash.
```
2017-10-16 20:45:22.7291 | ERROR | ParserHelper | Cannot parse below content. 
2017-10-16 20:45:22.7291 | ERROR | ParserHelper | <html>
<head><title>301 Moved Permanently</title></head>
<body bgcolor="white">
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

Adding the slash to the end of the URI has the server return the list and everything goes well after there. Copying from the Firefox URI bar will give the correct URI with the slash.

**Comment by moax101** [comment]
i cant download again in mangafox
![image](https://user-images.githubusercontent.com/32432153/33664061-1ab944b4-da47-11e7-9991-40a6e1b348ac.png)
its happening again

**Comment by moax101** [comment]
please help me

**Comment by lozowojo** [comment]
The site logged me out of my account and i can't connect anymore even tho i still have my username and password help me please. Btw the site changed from mangafox.me to mangafox.la does it change something ?
